Fix Vale violations across the Kedro documentation (#3293)
kedro-org/kedro: docs/source/data/data_catalog.md

```
### Dataset `type`

> Kedro supports a range of connectors, for CSV files, Excel spreadsheets, Parquet files, Feather files, HDF5 files, JSON documents, pickled objects, SQL tables, SQL queries, and more. They are supported using libraries such as pandas, PySpark, NetworkX, and Matplotlib.

Author: ```suggestion
Kedro supports a range of connectors, for CSV files, Excel spreadsheets, Parquet files, Feather files, HDF5 files, JSON documents, pickled objects, SQL tables, SQL queries and more. They are supported using libraries such as pandas, PySpark, NetworkX and Matplotlib.
```
Remove Oxford commas?

Reviewer: Let's leave them! I am (relatively) happy to use them for clarity and as Vale spots them missing throughout the docs in weeks to come, I'll add them in rather than take them out.

> Kedro supports a range of connectors, for CSV files, Excel spreadsheets, Parquet files, Feather files, HDF5 files, JSON documents, pickled objects, SQL tables, SQL queries, and more. They are supported using libraries like pandas, PySpark, NetworkX, Matplotlib and more.

Reviewer: ```suggestion
Kedro supports a range of connectors, for CSV files, Excel spreadsheets, Parquet files, Feather files, HDF5 files, JSON documents, pickled objects, SQL tables, SQL queries, and more. They are supported using libraries such as pandas, PySpark, NetworkX, and Matplotlib.
```

Author: @stichbury For my knowledge, should there be an Oxford comma here? Wondering because I came across a lot of instances where Vale wanted to add an Oxford comma, but I assumed not using one was the British way.

Reviewer: I've been adopting the Oxford comma as Vale requests it. I am not a big fan of it but it does have value in tech writing for clarity so where it's missing, and flagged as absent, it is fine to add it in.

[The `kedro-datasets` package documentation](/kedro_datasets) contains a comprehensive list of all available file types.

### Dataset `filepath`

Kedro relies on [`fsspec`](https://filesystem-spec.readthedocs.io/en/latest/) to read and save data from a variety of data stores including local file systems, network file systems, cloud object stores, and Hadoop. When specifying a storage location in `filepath:`, you should provide a URL using the general form `protocol://path/to/data`.  If no protocol is provided, the local file system is assumed (which is the same as ``file://``).

The following protocols are available:

- **Local or Network File System**: `file://` - the local file system is default in the absence of any protocol, it also permits relative paths.
- **Hadoop File System (HDFS)**: `hdfs://user@server:port/path/to/data` - Hadoop Distributed File System, for resilient, replicated files within a cluster.
- **Amazon S3**: `s3://my-bucket-name/path/to/data` - Amazon S3 remote binary store, often used with Amazon EC2,
  using the library s3fs.

> - **S3 Compatible Storage**: `s3://my-bucket-name/path/_to/data` - e.g. MinIO, using the s3fs library.

Reviewer: ```suggestion
- **S3 Compatible Storage**: `s3://my-bucket-name/path/_to/data` - for example, MinIO, using the s3fs library.
```
